lib/web_ui/lib/src/engine/pointer_binding/event_position_helper.dart

```diff
@@ -33,6 +47,34 @@ ui.Offset computeEventOffsetToTarget(DomMouseEvent event, DomElement actualTarge
   return ui.Offset(event.offsetX, event.offsetY);
 }
 
+/// Computes the offsets for input nodes, which live outside of the shadowDOM.
+/// Since inputs can be transformed (scaled, translated, etc), we can't rely on
+/// `_computeOffsetRelativeToActualTarget` to calculate accurate coordinates, as
+/// it only handles the case where inputs are translated, but will have issues
+/// for scaled inputs (see: https://github.com/flutter/flutter/issues/125948).
+///
+/// We compute the offsets here by using the text input geometry data that is
+/// sent from the framework, which includes information on how to transform the
+/// underlying input element. We transform the `event.offset` points we receive
+/// using the values from the input's transform matrix.
+ui.Offset _computeOffsetForInputs(DomMouseEvent event, EditableTextGeometry inputGeometry) {
+  final DomElement targetElement = event.target! as DomHTMLElement;
+  final DomHTMLElement domElement = textEditing.strategy.activeDomElement;
+  assert(targetElement == domElement, 'The targeted input element must be the active input element');
+  final Float32List transformValues = inputGeometry.globalTransform;
+  assert(transformValues.length == 16);
+  final Matrix4 transform = Matrix4(
+    transformValues[0], transformValues[1], transformValues[2], transformValues[3],
+    transformValues[4], transformValues[5], transformValues[6], transformValues[7],
+    transformValues[8], transformValues[9], transformValues[10], transformValues[11],
+    transformValues[12], transformValues[13], transformValues[14], transformValues[15],
+  );
```

Comment:
Maybe this is simpler?
```suggestion
  final Matrix4 transform = Matrix4.fromFloat32List(transformValues);
```

Reply:
Definitely.  Good catch